scripts/dump-swagger.py

```diff
@@ -155,6 +155,7 @@ def edit_links(node, base_url):
 
         basePath = api['basePath']
         for path, methods in api["paths"].items():
+            path = "{}{}".format(basePath, path)
```

Comment:
Any reason not to stick `basePath` into `output` before this loop?

Reply:
Note that `basePath` here is the one from each YAML in `data/api/client-server/`. It is not the one we have for the whole file, which we already set at line 115. By itself this one doesn’t belong in `output`. It’s used to build the full path of each API `path`, so this turns e.g.:
* `/matrix/client` from `wellknown.yaml` into `/.well-known/matrix/client` (and in turn `GET /.well-known/matrix/client` in the playground, which becomes consistent with [the spec documentation](https://spec.matrix.org/v1.1/client-server-api/#getwell-knownmatrixclient) and makes the Try feature work)
* `/login` from `login.yaml` into `/_matrix/client/v3/login`
* etc.


One question I had while writing this patch was why `basePath` was set (as needed) before the loop but never used. I assume whoever wrote this meant to write the same thing I did, but it slipped through.

Reply:
I should however have written it this way:

```suggestion
            path = basePath + path
```

There is no reason to use `format` here. :man_facepalming:

Reply:
Accepting my own suggestion marked this conversation as outdated and I didn’t mean to close it. Are you satisfied with my above explanation or do you still have concerns?